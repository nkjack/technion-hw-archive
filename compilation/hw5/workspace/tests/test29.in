//order of params
int f1(int x){
	print("f1!\n");
	return x;
}

byte f2(byte x) {
	print("f2!\n");
	return x;
}

bool f3(bool c) {
	print("f3!\n");
	return c;
}

void callMe(int a, int c, byte d, bool e, bool f, byte g, int h, bool i) {
	print("call me maybe");
	printi(a);
	printi(c);
	printi(d);
	printi(g);
	printi(h);
	print("\n");
}

void main() {
	callMe(f1(2),f2(4b),f2(5b),f3(true),f3(false),f2(3b),f1(22),f3(false));
}