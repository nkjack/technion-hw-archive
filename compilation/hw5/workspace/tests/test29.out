Loaded: ./exceptions.s
f1!
f2!
f2!
f3!
f3!
f2!
f1!
f3!
call me maybe245322
